### PR TITLE
#133846505 Fix the unit tests on accounts module

### PR DESCRIPTION
**Pull request by brotich**
#### What does this PR do?
Fix incomplete tests on the `hc/accounts/test` folder
#### Description of Task to be completed?
The folder `hc/accounts` has its unit tests in `hc/accounts/tests`. Some of these tests are incomplete and need fixing and some asserts added.
#### How should this be manually tested?
After cloning the repo and performing the set up according to the ReadMe
- `python manage.py test hc.accounts.tests`
#### Any background context you want to provide?
The files in the `hc/accounts/tests` contain incomplete tests described with lines starting with `##`
#### What are the relevant pivotal tracker stories? #133846505

### CODE REVIEW

File: hc/accounts/tests/test_check_token.py [hunk 1]
```diff
@@ -17,12 +17,25 @@ def test_it_redirects(self):
...
-        # After login, token should be blank
+        ### After login, token should be blank
         self.profile.refresh_from_db()
         self.assertEqual(self.profile.token, "")
 
-    ### Login and test it redirects already logged in
+        ### Login and test it redirects already logged in
 
-    ### Login with a bad token and check that it redirects
+        self.client.login(username="alice@example.org", password="password")
```
Comment (uzoawili) [inline hc/accounts/tests/test_check_token.py]: Consider moving the _### Login and test it redirects already logged in_ to a separate test.

File: hc/accounts/tests/test_login.py [hunk 2]
```diff
@@ -19,14 +19,20 @@ def test_it_sends_link(self):
...
+        ### Assert that a user was created        
+        self.assertEqual(User.objects.count(), 1)
+        
+                
+        ### And email sent
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(mail.outbox[0].subject, 'Log in to healthchecks.io')
+
         ### Assert contents of the email body
+        self.assertIn("To log into healthchecks.io, please open the link below:", mail.outbox[0].body)
```
Comment (uzoawili) [inline hc/accounts/tests/test_login.py:40]: In testing the content of the email body, consider testing that the correct login link is present too and not just the 'copy' string. The copy might be correct but the link in the email might be missing or wrongly formatted. ¯\_(ツ)_/¯

File: hc/accounts/tests/test_profile.py [hunk 3]
```diff
@@ -28,23 +33,33 @@ def test_it_sends_report(self):
...
+        assert len(mail.outbox) > 0
+
+        self.assertIn("alice@example.org", mail.outbox[0].to)
+        self.assertIn("Monthly Report", mail.outbox[0].subject)
+        self.assertIn("This is a monthly report sent by ", mail.outbox[0].body)
 
     def test_it_adds_team_member(self):
         self.client.login(username="alice@example.org", password="password")
 
         form = {"invite_team_member": "1", "email": "frank@example.org"}
```
Comment (uzoawili) [inline hc/accounts/tests/test_profile.py]: In the spirit of proper variable naming, `form_data` or simply `data` might be a more apt name to use here, given that forms have a specific meaning in Django

File: hc/accounts/tests/test_profile.py [hunk 4]
```diff
@@ -108,3 +123,16 @@ def test_it_shows_badges(self):
         self.assertNotContains(r, "bobs-tag.svg")
 
     ### Test it creates and revokes API key
+    def test_it_creates_api_key(self):
+        self.client.login(username="alice@example.org", password="password")
+        form = {'create_api_key': ''}
```
Comment (uzoawili) [inline hc/accounts/tests/test_profile.py]: Refer to previous comment on using `form` as the variable name in this context and other similar.

**Review by uzoawili — CHANGES_REQUESTED** [review]

File: hc/accounts/tests/test_profile.py (hunk 4 above)
Comment (brotich) [reply hc/accounts/tests/test_profile.py]: @andela-uawili noted the changes...will make them and update ASAP

File: hc/accounts/tests/test_profile.py (hunk 3 above)
Comment (brotich) [reply hc/accounts/tests/test_profile.py]: The variable names have been renamed following the convention: `form` renamed to `form_data` and `r` to `response` on specific files

**Review by uzoawili — APPROVED** [review]

**Pull request merged**